crates/matrix-sdk-ui/src/timeline/event_handler.rs

```diff
@@ -515,28 +509,35 @@ impl<'a, 'o> TimelineEventHandler<'a, 'o> {
         }
     }
 
-    /// If there's a pending edit for an item, applies it immediately, returning
-    /// an updated [`EventTimelineItem`]. Otherwise, return the original event
-    /// item.
-    fn maybe_apply_pending_edit(&mut self, item: &EventTimelineItem) -> Option<EventTimelineItem> {
-        let Flow::Remote { event_id, .. } = &self.ctx.flow else { return None };
-
-        if matches!(item.content(), TimelineItemContent::Message(..)) {
-            let pending = self.meta.pending_edits.remove(event_id)?;
-            let edit = as_variant!(pending, PendingEdit::RoomMessage)?;
-            return self.apply_msg_edit(item, edit);
-        }
-
-        if matches!(item.content(), TimelineItemContent::Poll(..)) {
-            let pending = self.meta.pending_edits.remove(event_id)?;
-            let edit = as_variant!(pending, PendingEdit::Poll)?;
-            return self.apply_poll_edit(item, edit);
+    /// If there's a pending edit for an item, apply it immediately, returning
+    /// an updated [`EventTimelineItem`]. Otherwise, return `None`.
+    fn maybe_unstash_pending_edit(
+        &mut self,
+        item: &EventTimelineItem,
+    ) -> Option<EventTimelineItem> {
+        if let Flow::Remote { event_id, .. } = &self.ctx.flow {
```

Comment:
Now you removed them completely, what was wrong with the suggestion I left? The suggestion left the first early return but got rid of the:

```
if return Some
if return Some

None
```

Reply:
I found it more consistent to only use early returns or not use any at all.